File: Curso_logica_de_programacao/Exercicios_logica_em_C++/Idades/Idades.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{

    string nome1, nome2;
    int idade1, idade2;
    double MediaIdade;

    cout << "Dados da primeira pessoa: \n";
    cout << "Nome: ";
    getline(cin, nome1);
    cout << "Idade: ";
    cin >> idade1;

    cout << "Dados da primeira pessoa: \n";
    cout << "Nome: ";
    cin.ignore(INT_MAX, '\n'); 
    getline(cin, nome2);
    cout << "Idade: ";
    cin >> idade2;

    MediaIdade = (double)(idade1 + idade2) / 2;

    cout << "A idade media de " << nome1 << " e " << nome2 << "e de " << MediaIdade << " anos";

    return 0;
}

File: Curso_logica_de_programacao/Exercicios_logica_em_Java/Idades/src/Idades.java
import java.util.Locale;
import java.util.Scanner;

public class Idades {

        public static void main(String[] args){

                Locale.setDefault(Locale.US);
                Scanner sc = new Scanner(System.in);

                System.out.println("Dados da primeira pessoa: ");
                System.out.print("Nome: ");
                String nome1 = sc.nextLine();
                System.out.print("Idade: ");
                int idade1 = sc.nextInt();
                sc.nextLine();
                System.out.println("Dados da segunda pessoa: ");
                System.out.print("Nome: ");
                String nome2 = sc.nextLine();
                System.out.print("Idade: ");
                int idade2 = sc.nextInt();

                double media_idade = (double) (idade1 + idade2) / 2.0;

                System.out.print("A idade media de " + nome1 + " e " + nome2 + " e de " + String.format("%.1f", media_idade) + " anos");

                sc.close();
        }
}
